R/operators.R

```diff
@@ -1079,22 +1079,23 @@ path <- function(...) {
   } else if ("igraph.vertex" %in% class(e2)) {
     ## Adding vertices, possibly with attributes
     ## If there is a single unnamed argument, that contains the vertex names
-    wn <- which(names(e2) == "")
-    if (length(wn) == 1) {
-      names(e2)[wn] <- "name"
+    unnamed_elements_indices <- which(!rlang::have_name(e2))
+    if (length(unnamed_elements_indices) == 1) {
+      e2 <- rlang::set_names(e2[unnamed_elements_indices], "name")
     } else if (is.null(names(e2))) {
       ## No names at all, everything is a vertex name
       e2 <- list(name = unlist(e2, recursive = FALSE))
-    } else if (length(wn) == 0) {
+    } else if (length(unnamed_elements_indices) == 0) {
       ## If there are no non-named arguments, we are fine
     } else {
       ## Otherwise, all unnamed arguments are collected and used as
```

Comment:
this is the case that was broken as far as I understand. Not sure why it works to fix it here.